// @morev/utils/dist/index.js
import "./chunk-5DNDSZ5N.js";
import {
  minTimePromise
} from "./chunk-WVNRJQVA.js";
import {
  promiseController
} from "./chunk-PRKUJMFN.js";
import "./chunk-56HQMUPL.js";
import {
  rangesInvert
} from "./chunk-3WWQJ5QK.js";
import {
  rangeIncludes
} from "./chunk-UHI36KE4.js";
import {
  rangesCrop
} from "./chunk-B3HOJW7D.js";
import "./chunk-TTZZQRNK.js";
import {
  queryStringify
} from "./chunk-LU3UT6MV.js";
import "./chunk-OHMNL74K.js";
import {
  randomDate
} from "./chunk-P6P5WGWY.js";
import {
  daysInMonth
} from "./chunk-OI2ZBQLP.js";
import {
  createFormatDate,
  formatDate,
  formatDateEn,
  formatDateRu
} from "./chunk-I23E2RTO.js";
import {
  toDate
} from "./chunk-YTKBKGCJ.js";
import {
  parseDate
} from "./chunk-KFM23YRT.js";
import {
  FORMAT_DATE_LOCALE_EN,
  FORMAT_DATE_LOCALE_RU
} from "./chunk-M5HBLC4T.js";
import "./chunk-ZH3ZSQ4E.js";
import {
  isIsoDate
} from "./chunk-DYPEECVT.js";
import {
  isLeapYear
} from "./chunk-Q3HRT6I3.js";
import "./chunk-FADNZEVI.js";
import {
  getScrollbarWidth
} from "./chunk-J7NKYSCD.js";
import {
  getElement
} from "./chunk-HAKRSJIV.js";
import {
  getElementOffset
} from "./chunk-CTAWBVQ2.js";
import {
  getWindowScroll
} from "./chunk-RN2LZAEZ.js";
import {
  getScrollLimit
} from "./chunk-AZKEOTOA.js";
import {
  getDocumentSize
} from "./chunk-JKOWNRAJ.js";
import {
  getScrollableAncestor
} from "./chunk-QH2XCLML.js";
import "./chunk-2LYOD43H.js";
import {
  isMobile
} from "./chunk-L6JF55BK.js";
import {
  isMotionless
} from "./chunk-PYDMVANF.js";
import {
  isOSX
} from "./chunk-WCAQQIST.js";
import {
  isProd
} from "./chunk-XUMPAFUC.js";
import {
  isServer
} from "./chunk-TTSFIVKA.js";
import {
  isTest
} from "./chunk-PJOEEWLD.js";
import {
  isTouch
} from "./chunk-SEJGFAF3.js";
import {
  supportsPassive
} from "./chunk-MF6XUJNR.js";
import {
  isDev
} from "./chunk-MFLP4GWI.js";
import {
  isFirefox
} from "./chunk-TVMLY5SN.js";
import {
  isInternetExplorer
} from "./chunk-32TFFOFK.js";
import {
  isLighthouse
} from "./chunk-GTUT74YY.js";
import {
  isClient
} from "./chunk-CRYBHWGN.js";
import "./chunk-C4PFADZD.js";
import {
  sleep
} from "./chunk-HROAT5GA.js";
import {
  throttle
} from "./chunk-YBGX6WUZ.js";
import {
  verifyFileAccept
} from "./chunk-HUZXOUDA.js";
import {
  withProbability
} from "./chunk-CKYSU2BH.js";
import "./chunk-AKHMGIKW.js";
import {
  compose
} from "./chunk-BZUIRQQN.js";
import {
  debounce
} from "./chunk-XN4T4JSQ.js";
import {
  hash
} from "./chunk-57JKRZYH.js";
import {
  noop
} from "./chunk-MR7JPI4H.js";
import {
  safeJsonParse
} from "./chunk-4UM6RWHQ.js";
import "./chunk-TUPXTHDB.js";
import {
  lerp
} from "./chunk-R7SKROQY.js";
import {
  angle
} from "./chunk-B2ZKLV74.js";
import {
  distance
} from "./chunk-7DIUBCVI.js";
import {
  gcd
} from "./chunk-DUCEKAA2.js";
import "./chunk-6QGJD63N.js";
import {
  rgbToHex
} from "./chunk-LI7RBS2T.js";
import {
  contrastColor
} from "./chunk-AYLEOSRB.js";
import {
  hexToRgb
} from "./chunk-7IJ4CVFJ.js";
import {
  normalizeHex
} from "./chunk-B4YLZ5QK.js";
import "./chunk-4WFMUP35.js";
import {
  formatBytes,
  formatBytesRu
} from "./chunk-PRSBP434.js";
import {
  numberFormat
} from "./chunk-AVXWFG47.js";
import {
  randomFloat
} from "./chunk-EWXRMA6N.js";
import {
  precision
} from "./chunk-57V3ZFZB.js";
import "./chunk-5LECPUSN.js";
import {
  pick
} from "./chunk-ZXVPBFYX.js";
import {
  deepClone,
  deepCloneStrict
} from "./chunk-Q5PZZCDY.js";
import {
  deepEqual,
  deepEqualCircular
} from "./chunk-5YCTAAGH.js";
import {
  formDataToObject
} from "./chunk-XTKU53CL.js";
import {
  createMergeObjects,
  mergeObjects
} from "./chunk-AQ4NUHLD.js";
import {
  objectGet
} from "./chunk-TACI4XVG.js";
import {
  objectKeysCase
} from "./chunk-4KQODNHI.js";
import {
  objectToFormdata
} from "./chunk-ZXQ2DRQF.js";
import {
  tsObject
} from "./chunk-K2EWJMCD.js";
import {
  omit
} from "./chunk-WFLW3WOP.js";
import {
  toNumber
} from "./chunk-4MABVGSG.js";
import "./chunk-BODHRZ2V.js";
import "./chunk-PSAKS244.js";
import {
  serializeCookie
} from "./chunk-SI2ZEYB5.js";
import {
  parseResponseCookie
} from "./chunk-ACQE26QV.js";
import "./chunk-RZDOTCIA.js";
import "./chunk-OSBSRYIM.js";
import {
  unquote
} from "./chunk-OIOEMUWT.js";
import {
  padEnd
} from "./chunk-DENZN3B2.js";
import {
  padStart
} from "./chunk-WDLKPA7O.js";
import {
  pascalCase
} from "./chunk-A2T2JGZ2.js";
import {
  quote
} from "./chunk-N26I5ZBS.js";
import {
  randomString
} from "./chunk-ZFEL4F7H.js";
import {
  romanize
} from "./chunk-CGRZQPAE.js";
import {
  snakeCase
} from "./chunk-7SVUEZ2T.js";
import {
  stripIndent
} from "./chunk-EF6O6L7Q.js";
import {
  camelCase
} from "./chunk-FKKK4IRM.js";
import {
  capitalize
} from "./chunk-CGEQLLGR.js";
import {
  escapeRegExp
} from "./chunk-4APQF3QV.js";
import {
  fileExtension
} from "./chunk-AMA5DX26.js";
import {
  formatSlashes
} from "./chunk-MLONGT6L.js";
import {
  kebabCase
} from "./chunk-IC7PXDZS.js";
import "./chunk-4UVMFHM7.js";
import {
  arraysDifference
} from "./chunk-XIEVB4AG.js";
import {
  arraysIntersection
} from "./chunk-4TKPKUM3.js";
import {
  arraysUnion
} from "./chunk-XXCXULIL.js";
import {
  asyncArray
} from "./chunk-7BEVQKDU.js";
import "./chunk-WMCN5LBL.js";
import {
  arrayInsert,
  arrayInsertMutable
} from "./chunk-XYOXRCP6.js";
import {
  arrayRange
} from "./chunk-QSUEZT35.js";
import {
  arrayRemove,
  arrayRemoveMutable
} from "./chunk-MNX5XL3V.js";
import {
  arraySample
} from "./chunk-J63LHLH4.js";
import {
  randomInteger
} from "./chunk-FDBXWY5V.js";
import {
  arrayOfLength
} from "./chunk-36MHHFDE.js";
import {
  arrayShuffle
} from "./chunk-IECQATAF.js";
import {
  arraySwap,
  arraySwapMutable
} from "./chunk-RZNP6SFE.js";
import {
  clamp
} from "./chunk-EJGCYSTB.js";
import {
  toArray
} from "./chunk-OAZLYBJA.js";
import {
  arrayUnique
} from "./chunk-FER4ZC5Z.js";
import {
  rangesIntersection
} from "./chunk-7DO2V6XN.js";
import {
  rangesMerge
} from "./chunk-W7KSBM75.js";
import {
  rangesSort
} from "./chunk-VAXS4RSF.js";
import {
  assert,
  configureAssert
} from "./chunk-MHPJZ3EK.js";
import "./chunk-G3ENWOTI.js";
import "./chunk-ZX6GOGJL.js";
import "./chunk-BT72POBR.js";
import {
  isSessionStorageAvailable
} from "./chunk-O5DS3PCZ.js";
import {
  isSymbol
} from "./chunk-TKM2FQTA.js";
import {
  isWeakMap
} from "./chunk-Y3T3BIRX.js";
import {
  isPlainObject
} from "./chunk-VGTXDYHX.js";
import {
  isPrimitive
} from "./chunk-BQOAKWXA.js";
import {
  isNullish
} from "./chunk-Z5IVBZLV.js";
import {
  isUndefined
} from "./chunk-F45GO7K2.js";
import {
  isNull
} from "./chunk-TK4YRMRC.js";
import {
  isPromise
} from "./chunk-SPK7OES6.js";
import {
  isRegExp
} from "./chunk-5HZ32HEL.js";
import {
  isFormData
} from "./chunk-ON2BNX4R.js";
import {
  isFunction
} from "./chunk-CQJCSTBU.js";
import {
  isHex
} from "./chunk-NTE2FZDH.js";
import {
  isInteger
} from "./chunk-4W3EM56D.js";
import {
  isLocalStorageAvailable
} from "./chunk-SIR5WJCA.js";
import {
  isMap
} from "./chunk-EXCPRRER.js";
import {
  isNode
} from "./chunk-M3TQVFXR.js";
import {
  isDate
} from "./chunk-UOWY6RIU.js";
import {
  isEmail
} from "./chunk-3SNQM74H.js";
import {
  isEmpty
} from "./chunk-MYOGSMXY.js";
import {
  isObject
} from "./chunk-XUWL4VKZ.js";
import {
  isIterable
} from "./chunk-HJKYC4E4.js";
import {
  isError
} from "./chunk-WE7QC4OC.js";
import {
  isFile
} from "./chunk-PDZUYQZ5.js";
import {
  isFloat
} from "./chunk-MV2MTLNA.js";
import {
  isNumeric
} from "./chunk-OBG63REY.js";
import {
  isArray
} from "./chunk-UDVSPJHY.js";
import {
  isBlob
} from "./chunk-FXAYJFKY.js";
import {
  isBoolean
} from "./chunk-42JGV6MK.js";
import {
  isElement
} from "./chunk-FMEA3XAZ.js";
import {
  isString
} from "./chunk-RUB2ELNH.js";
import "./chunk-G4Y5YQL7.js";
export {
  FORMAT_DATE_LOCALE_EN,
  FORMAT_DATE_LOCALE_RU,
  angle,
  arrayInsert,
  arrayInsertMutable,
  arrayOfLength,
  arrayRange,
  arrayRemove,
  arrayRemoveMutable,
  arraySample,
  arrayShuffle,
  arraySwap,
  arraySwapMutable,
  arrayUnique,
  arraysDifference,
  arraysIntersection,
  arraysUnion,
  assert,
  asyncArray,
  camelCase,
  capitalize,
  clamp,
  compose,
  configureAssert,
  contrastColor,
  createFormatDate,
  createMergeObjects,
  daysInMonth,
  debounce,
  deepClone,
  deepCloneStrict,
  deepEqual,
  deepEqualCircular,
  distance,
  escapeRegExp,
  fileExtension,
  formDataToObject,
  formatBytes,
  formatBytesRu,
  formatDate,
  formatDateEn,
  formatDateRu,
  formatSlashes,
  gcd,
  getDocumentSize,
  getElement,
  getElementOffset,
  getScrollLimit,
  getScrollableAncestor,
  getScrollbarWidth,
  getWindowScroll,
  hash,
  hexToRgb,
  isArray,
  isBlob,
  isBoolean,
  isClient,
  isDate,
  isDev,
  isElement,
  isEmail,
  isEmpty,
  isError,
  isFile,
  isFirefox,
  isFloat,
  isFormData,
  isFunction,
  isHex,
  isInteger,
  isInternetExplorer,
  isIsoDate,
  isIterable,
  isLeapYear,
  isLighthouse,
  isLocalStorageAvailable,
  isMap,
  isMobile,
  isMotionless,
  isNode,
  isNull,
  isNullish,
  isNumeric,
  isOSX,
  isObject,
  isPlainObject,
  isPrimitive,
  isProd,
  isPromise,
  isRegExp,
  isServer,
  isSessionStorageAvailable,
  isString,
  isSymbol,
  isTest,
  isTouch,
  isUndefined,
  isWeakMap,
  kebabCase,
  lerp,
  mergeObjects,
  minTimePromise,
  noop,
  normalizeHex,
  numberFormat,
  objectGet,
  objectKeysCase,
  objectToFormdata,
  omit,
  padEnd,
  padStart,
  parseDate,
  parseResponseCookie,
  pascalCase,
  pick,
  precision,
  promiseController,
  queryStringify,
  quote,
  randomDate,
  randomFloat,
  randomInteger,
  randomString,
  rangeIncludes,
  rangesCrop,
  rangesIntersection,
  rangesInvert,
  rangesMerge,
  rangesSort,
  rgbToHex,
  romanize,
  safeJsonParse,
  serializeCookie,
  sleep,
  snakeCase,
  stripIndent,
  supportsPassive,
  throttle,
  toArray,
  toDate,
  toNumber,
  tsObject,
  unquote,
  verifyFileAccept,
  withProbability
};
